subprojects/robotpy-wpilib/wpilib/iterativerobotpy.py

```diff
@@ -0,0 +1,237 @@
+    kTeleop = 3
+    kTest = 4
+
+
+class IterativeRobotPy(wpilib.RobotBase):
+
+    def __init__(self, period: float):
+        super().__init__()
+        self._m_word = wpilib.DSControlWord()
+        self._m_period = period
+        self._m_watchdog = wpilib.Watchdog(self._m_period, self.printLoopOverrunMessage)
+        self._mode: IterativeRobotMode = IterativeRobotMode.kNone
+        self._m_lastMode: IterativeRobotMode = IterativeRobotMode.kNone
+        self._m_ntFlushEnabled: bool = True
+        self._m_lwEnabledInTest: bool = False
+        self._m_calledDsConnected: bool = False
+        self._m_reportedLw: bool = False
+        self._robotPeriodicHasRun: bool = False
+        self._simulationPeriodicHasRun: bool = False
+        self._disabledPeriodicHasRun: bool = False
+        self._autonomousPeriodicHasRun: bool = False
+        self._teleopPeriodicHasRun: bool = False
+        self._testPeriodicHasRun: bool = False
+
+    def printLoopOverrunMessage(self):
+        pass
+
+    def robotInit(self):
+        pass
+
+    def driverStationConnected(self):
+        pass
+
+    def simulationInit(self):
+        pass
+
+    def disabledInit(self):
+        pass
+
+    def autonomousInit(self):
+        pass
+
+    def teleopInit(self):
+        pass
+
+    def testInit(self):
+        pass
+
+    def robotPeriodic(self):
+        if not self._robotPeriodicHasRun:
+            print(f"Default RobotPeriodic() method...Override me!")
+            self._robotPeriodicHasRun = True
+
+    def simulationPeriodic(self):
+        if not self._simulationPeriodicHasRun:
+            print(f"Default simulationPeriodic() method...Override me!")
+            self._simulationPeriodicHasRun = True
+
+    def disabledPeriodic(self):
+        if not self._disabledPeriodicHasRun:
+            print(f"Default disabledPeriodic() method...Override me!")
+            self._disabledPeriodicHasRun = True
+
+    def autonomousPeriodic(self):
+        if not self._autonomousPeriodicHasRun:
+            print(f"Default autonomousPeriodic() method...Override me!")
+            self._autonomousPeriodicHasRun = True
+
+    def teleopPeriodic(self):
+        if not self._teleopPeriodicHasRun:
+            print(f"Default teleopPeriodic() method...Override me!")
+            self._teleopPeriodicHasRun = True
+
+    def testPeriodic(self):
+        if not self._testPeriodicHasRun:
+            print(f"Default testPeriodic() method...Override me!")
+            self._teleopPeriodicHasRun = True
+
+    def disabledExit(self):
+        pass
+
+    def autonomousExit(self):
+        pass
+
+    def teleopExit(self):
+        pass
+
+    def testExit(self):
+        pass
+
+    # todo @Deprecated(forRemoval=true, since="2025")
+    def setNetworkTablesFlushEnabled(self, enabled: bool):
+        self._m_ntFlushEnabled = enabled
+
+    def enableLiveWindowInTest(self, testLW: bool):
+        if self.isTestEnabled():
+            pass
+            # todo throw
```

Comment:
At the moment there isn't a binding for FRC_MakeError. For now, I'd say let's just add an `errors.py` and add an `class FRCError(RuntimeError):` exception type and use that.

Reply:
I think in this case we can just use RuntimeError directly? I wouldn't expect any teams to try to catch this exception.